keys.go

```diff
@@ -133,103 +133,96 @@ type KeyVersion struct {
-	return c.CreateImportedKey(ctx, name, expiration, payload, "", "", true)
+func WithAliases(aliases []string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Aliases = aliases
+		return "aliases"
+	}
 }
 
-// CreateKeyWithAliaes creats a new key with alias names. A key can have a maximum of 5 alias names.
-// For more information please refer to the links below:
-// https://cloud.ibm.com/docs/key-protect?topic=key-protect-create-root-keys#create-root-key-api
-// https://cloud.ibm.com/docs/key-protect?topic=key-protect-create-standard-keys#create-standard-key-api
-func (c *Client) CreateKeyWithAliases(ctx context.Context, name string, expiration *time.Time, extractable bool, aliases []string) (*Key, error) {
-	return c.CreateImportedKeyWithAliases(ctx, name, expiration, "", "", "", extractable, aliases)
+func WithPolicy(policy *Policy) CreateKeyOption {
+	return func(key *Key) string {
+		key.Rotation = policy.Rotation
+		key.DualAuthDelete = policy.DualAuth
+		return "policy"
+	}
 }
 
-// CreateImportedKeyWithAliases creates a new key with alias name and provided key material. A key can have a maximum of 5 alias names
-// When importing root keys with import-token encryptedNonce and iv need to passed along with payload.
-// Standard Keys cannot be imported with an import token hence only payload is required.
-// For more information please refer to the links below:
-// https://cloud.ibm.com/docs/key-protect?topic=key-protect-import-root-keys#import-root-key-api
-// https://cloud.ibm.com/docs/key-protect?topic=key-protect-import-standard-keys#import-standard-key-gui
-func (c *Client) CreateImportedKeyWithAliases(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string, extractable bool, aliases []string) (*Key, error) {
-	key := c.createKeyTemplate(ctx, name, expiration, payload, encryptedNonce, iv, extractable, aliases, AlgorithmRSAOAEP256, nil)
-	return c.createKey(ctx, key)
+func WithTags(tags []string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Tags = tags
+		return "tags"
+	}
 }
 
-func (c *Client) createKeyTemplate(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string, extractable bool, aliases []string, encryptionAlgorithm string, policy *Policy) Key {
-	key := Key{
+func (c *Client) CreateKey(ctx context.Context, name string, extractable bool, options ...CreateKeyOption) (*Key, error) {
+	key := &Key{
 		Name:        name,
 		Type:        keyType,
 		Extractable: extractable,
-		Payload:     payload,
 	}
-
-	if aliases != nil {
-		key.Aliases = aliases
-	}
-
-	if !extractable && payload != "" && encryptedNonce != "" && iv != "" {
-		key.EncryptedNonce = encryptedNonce
-		key.IV = iv
-		key.EncryptionAlgorithm = encryptionAlgorithm
-	}
-
-	if expiration != nil {
-		key.Expiration = expiration
+	overrides := false
+	for _, opt := range options {
+		if opt != nil {
+			optName := opt(key)
```

Comment:
for each of the `WithX` functions, they will execute on the key struct in here, adding the fields as needed.